tests/evaluate/test_evaluation.py

```diff
@@ -1567,3 +1568,257 @@ def test_evaluate_with_static_dataset_error_handling_pandas_dataset():
+        (
+            pd.DataFrame(
+                {
+                    "inputs": [
+                        {
+                            "messages": [{"content": q, "role": "user"}],
+                            "max_tokens": 10,
+                        }
+                        for q in _TEST_QUERY_LIST
+                    ],
+                    "ground_truth": _TEST_GT_LIST,
+                }
+            ),
+            None,
+            "ground_truth",
+        ),
+        # List of string
+        (
+            [[q] for q in _TEST_QUERY_LIST],
+            None,
+            _TEST_GT_LIST,
+        ),
+        # List of string with feature_names
+        (
+            [[q] for q in _TEST_QUERY_LIST],
+            ["question"],
+            _TEST_GT_LIST,
+        ),
+        # List of string with feature_names and w/o targets
+        (
+            [[q] for q in _TEST_QUERY_LIST],
+            ["question"],
+            None,
+        ),
+        # List of dictionary with feature_names
+        (
+            [
+                [
+                    {
+                        "messages": [{"content": q, "role": "user"}],
+                        "max_tokens": 10,
+                    }
+                ]
+                for q in _TEST_QUERY_LIST
+            ],
+            None,
+            _TEST_GT_LIST,
+        ),
+    ],
+)
+@mock.patch("mlflow.deployments.get_deploy_client")
+def test_evaluate_on_chat_model_endpoint(mock_deploy_client, input_data, feature_names, targets):
+    mock_deploy_client.return_value.predict.return_value = _DUMMY_CHAT_RESPONSE
+    mock_deploy_client.return_value.get_endpoint.return_value = {"task": "llm/v1/chat"}
+
+    with mlflow.start_run():
+        eval_result = mlflow.evaluate(
+            model="endpoints:/chat",
+            data=input_data,
+            model_type="question-answering",
+            feature_names=feature_names,
+            targets=targets,
+            inference_params={"max_tokens": 10, "temperature": 0.5},
+        )
+
+    call_args_list = mock_deploy_client.return_value.predict.call_args_list
+    expected_calls = [
+        mock.call(
+            endpoint="chat",
+            inputs={
+                "messages": [{"content": "What is MLflow?", "role": "user"}],
+                "max_tokens": 10,
+                "temperature": 0.5,
+            },
+        ),
+        mock.call(
+            endpoint="chat",
+            inputs={
+                "messages": [{"content": "What is Spark?", "role": "user"}],
+                "max_tokens": 10,
+                "temperature": 0.5,
+            },
+        ),
+    ]
+    assert all(call in call_args_list for call in expected_calls)
+    expected_metrics_subset = {"toxicity/v1/ratio", "ari_grade_level/v1/mean"}
+    if targets:
+        expected_metrics_subset.add("exact_match/v1")
+    assert expected_metrics_subset.issubset(set(eval_result.metrics.keys()))
```

Comment:
Can we also add some asserts on the contents of the per-row results / metric logs from `eval_results_table`? Specifically, I'd like to make sure that the model output column for an endpoint only contains the `content` of the first `message`, not the full `_DUMMY_COMPLETION_RESPONSE`.

Reply:
That's a great idea!